client/state/themes/selectors.js

```diff
@@ -417,10 +417,9 @@ export function getThemeHelpUrl( state, themeId, siteId ) {
  * @return {?String}         Theme purchase URL
  */
 export function getThemePurchaseUrl( state, themeId, siteId ) {
-	if ( isJetpackSite( state, siteId ) || ! isThemePremium( state, themeId ) ) {
+	if ( ! isThemePremium( state, themeId ) ) {
```

Comment:
I think we should drop this change -- there still isn't such a thing as a purchase URL for a singular premium theme on a JP site.